### ISSUE TITLE
Getting Array instead of DeviceArray  

### ISSUE THREAD

**Issue by tttttanya** [google/jax#17935]
### Description

Hi Team,

I am trying to understand the `Asynchronous dispatch` [link](https://jax.readthedocs.io/en/latest/async_dispatch.html) 
and when executing the below program (in colab) mentioned in this page I am getting `Array` instead of `DeviceArray`.
<img width="739" alt="Screenshot 2023-10-04 at 2 19 53 PM" src="https://github.com/google/jax/assets/118018621/de3f3808-945d-4b68-aa0d-b97c5435107e">

My query is whats the difference between `Array` and `DeviceArray`?
Secondly, do we need document update here?

```
import numpy as np
import jax.numpy as jnp
from jax import random
x = random.uniform(random.PRNGKey(0), (1000, 1000))
# Printing the result (i.e. evaluating `repr(result)` or `str(result)`)
# will block until the value is ready.
jnp.dot(x, x) + 3. 
```

### What jax/jaxlib version are you using?

jax    0.4.16 ,    jaxlib   0.4.16+cuda11.cudnn86

### Which accelerator(s) are you using?

GPU

### Additional system info

Mac

**Comment by yashk2810** [comment]
We need to update that doc but DeviceArray was deleted in favor of Array. https://jax.readthedocs.io/en/latest/jax_array_migration.html goes into more details.

**Comment by hawkinsp** [comment]
I think we can declare this fixed not #17965 was merged.
